powersync-ja/powersync-js
# feat(attachments): add error handlers

## Description
Add handlers to allow user to deal with `uploads` and `downloads`

## Work Done
* Added `onDownloadError` and `onUploadError` optional options 
* Updated demo app to showcase an example of how to use it

## How to test
It's easiest to test this through the demo todo-list app.

* Delete an attachment from supabase bucket that is connected to a todo item you have
* Run the app
* You should not see an error for the download
* Go to `system.ts` and change the `{ retry: true }` in `onDownloadError` to `{ retry: false }`. Reload the app, you should see the error
